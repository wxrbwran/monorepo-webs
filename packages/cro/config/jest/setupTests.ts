// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});
