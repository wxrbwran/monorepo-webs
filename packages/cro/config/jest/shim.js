import 'core-js/stable';
import 'regenerator-runtime/runtime';

/* eslint-disable import/first */
global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function requestAnimationFrame(callback) {
    setTimeout(callback, 0);
  };

global.matchMedia = window.matchMedia = () => ({
  matches: false,
  addListener: () => { },
  removeListener: () => { },
});
