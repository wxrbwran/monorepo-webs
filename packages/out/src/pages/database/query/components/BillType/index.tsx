import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Select, Checkbox, DatePicker, Button } from 'antd';
import moment from 'moment';
import styles from '../../index.scss';

const { Option } = Select;
const { RangePicker } = DatePicker;

interface IProps {
  laboratoryType: { key: string; value: string }[];
  toggleTab: () => void;
}

const timeType = [
  {
    key: 'ALL',
    value: '全部',
  },
  {
    key: 'CUSTOM',
    value: '自定义',
  },
];

function BillType(props: IProps) {
  const dispatch = useDispatch();
  const { laboratoryType, toggleTab } = props;
  const [type, changeType] = useState({}); // {SHQX:'CUSTOM', XCG: 'ALL'}
  const [imgkey, setImgkey] = useState<string[]>([]); // ['SHQX','XCG','BCG']
  const [reportData, setReportData] = useState({}); // {SHQX: {endAt: '', startAt: '',imageType: ''}, XCG: {}}
  const images = useSelector((state: IQuery) => state.query.images);

  const changeVal = (e: any, key: string) => {
    const isChecked = e.target.checked; // 选中取消选中状态
    if (isChecked) {
      setImgkey([...imgkey, key]);
      setReportData({
        ...reportData,
        [key]: {
          imageType: key,
          startAt: null,
          endAt: null,
        },
      });
    } else {
      setImgkey(imgkey.filter((item) => item !== key));
      delete type[key];
      delete reportData[key];
    }
  };

  const handleGetType = (value: string, key: string) => {
    changeType({ ...type, [key]: value });
  };

  const changeDate = (_date: any, dateString: string[], key: string) => {
    // console.log('date', date, dateString, key);
    setReportData({
      ...reportData,
      [key]: {
        startAt: moment(dateString[0]).valueOf(),
        endAt: moment(dateString[1]).valueOf(),
        imageType: key,
      },
    });
  };

  const onCancel = () => {
    changeType({});
    setImgkey([]);
    setReportData({});
    toggleTab();
  };

  const onOk = () => {
    const val = [...Object.values(reportData), ...images];
    const obj: CommonData = {};
    const newImages = val.reduce((cur, next) => {
      if (!obj[next.imageType]) {
        obj[next.imageType] = true;
        cur.push(next);
      }
      return cur;
    }, []);
    toggleTab();
    dispatch({
      type: 'query/setImages',
      payload: newImages,
    });
  };

  return (
    <div className={styles.indicator_wrapper}>
      {laboratoryType.map((item) => (
        <div className={styles.indicator} key={item.key}>
          <Checkbox onChange={(e) => changeVal(e, item.key)}>{item.value}</Checkbox>
          {imgkey.includes(item.key) && (
            <>
              <div className={styles.time_type}>
                <span className={styles.label}>时间：</span>
                <Select
                  style={{ width: 91 }}
                  onChange={(value) => handleGetType(value, item.key)}
                  defaultValue="ALL"
                >
                  {timeType.map((vitem) => (
                    <Option value={vitem.key} key={vitem.key}>
                      {vitem.value}
                    </Option>
                  ))}
                </Select>
              </div>
              {type[item.key] === 'CUSTOM' && (
                <RangePicker
                  onChange={(date, dateString) => changeDate(date, dateString, item.key)}
                />
              )}
            </>
          )}
        </div>
      ))}
      <div className={styles.save_btn}>
        <Button onClick={onCancel}>取消</Button>
        <Button type="primary" onClick={onOk}>
          {' '}
          确定{' '}
        </Button>
      </div>
    </div>
  );
}

export default BillType;
