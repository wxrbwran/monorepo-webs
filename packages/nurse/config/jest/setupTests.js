// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import 'jest';
// import axios from 'axios';
// import '@testing-library/jest-dom/extend-expect';

// Enzyme.configure({ adapter: new Adapter() });

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});

// jest.mock('@/services/api', () => {
//     const API = {
//         get: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         post: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         setAuthorizationToken: jest.fn(),
//     };
//     const http = API;
//     return http;
// })
