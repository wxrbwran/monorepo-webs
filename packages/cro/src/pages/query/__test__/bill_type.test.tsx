import 'jest';
import React from 'react';
import BillType from '../components/bill_type';
import { render, RenderResult, fireEvent, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({
  query: {
    images: [],
    other: [],
    base: {}
  },
});
const laboratoryType = [
  {
    key: 'SHQX',
    value: '生化全项',
  }, {
    key: 'XCG',
    value: '血常规',
  }, {
    key: 'BCG',
    value: '便常规',
  }
]

const checkType = [
  {
    key: 'XZCS',
    value: '超声',
  }, {
    key: 'XDT',
    value: '心电图',
  }
]
const laboratoryPorps = {
  laboratoryType: laboratoryType,
  toggleTab: jest.fn(),
  billType: 'LABORATORY'
}
const checkPorps = {
  laboratoryType: checkType,
  toggleTab: jest.fn(),
  billType: 'CHECK'
}
describe('查询页面：化验单选择组件', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = render(<Provider store={store}><BillType {...laboratoryPorps} /></Provider>)
  })
  it('匹配快照', () => {
    expect(wrapper).toMatchSnapshot();
    // wrapper.debug();
  })
  // it('勾选化验单类型后，显示时间选择框', async() => {
  //   const checkbox = wrapper.getByTestId('checkbox0');
  //   fireEvent.change(checkbox, {
  //     target: { checked: true }
  //   })
  //   await waitFor(() => {
  //     // expect(wrapper.getByTestId('time0')).toBeVisible();
  //   })
  // })
})
// describe('查询页面：检查单选择组件', () => {
//   let wrapper: RenderResult;
//   beforeEach(() => {
//     wrapper = render(<Provider store={store}><BillType {...checkPorps} /></Provider>)
//   })
// })