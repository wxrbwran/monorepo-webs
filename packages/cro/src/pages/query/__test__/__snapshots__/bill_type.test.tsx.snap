// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`查询页面：化验单选择组件 匹配快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                data-testid="checkbox0"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              生化全项
            </span>
          </label>
        </div>
        <div>
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                data-testid="checkbox1"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              血常规
            </span>
          </label>
        </div>
        <div>
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                data-testid="checkbox2"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              便常规
            </span>
          </label>
        </div>
        <div>
          <button
            class="ant-btn"
            type="button"
          >
            <span>
              取 消
            </span>
          </button>
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
               确定 
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <label
          class="ant-checkbox-wrapper"
        >
          <span
            class="ant-checkbox"
          >
            <input
              class="ant-checkbox-input"
              data-testid="checkbox0"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span>
            生化全项
          </span>
        </label>
      </div>
      <div>
        <label
          class="ant-checkbox-wrapper"
        >
          <span
            class="ant-checkbox"
          >
            <input
              class="ant-checkbox-input"
              data-testid="checkbox1"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span>
            血常规
          </span>
        </label>
      </div>
      <div>
        <label
          class="ant-checkbox-wrapper"
        >
          <span
            class="ant-checkbox"
          >
            <input
              class="ant-checkbox-input"
              data-testid="checkbox2"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span>
            便常规
          </span>
        </label>
      </div>
      <div>
        <button
          class="ant-btn"
          type="button"
        >
          <span>
            取 消
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary"
          type="button"
        >
          <span>
             确定 
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
